### PR TITLE
add swerve and sushiswap to FQT

### PR DESCRIPTION
**Pull request by xianny**
## Description

Add Sushiswap and Swerve to FillQuoteTransformer.

* Add SwerveBridge contract (duplicate of CurveBridge, we make them distinct for data tracking)
* Add MixinSushiswap contract
* Add `swerveBridge`, `sushiswapBridge`, and `sushiswapV2Router` fields to adapter addresses.

TODO
- [ ] Deploy SwerveBridge
- [x] Deploy SushiswapBridge (@dekz is this done?)
- [ ] Redeploy BridgeAdapter with new addresses. Sushiswap Router is here: https://etherscan.io/address/0xd9e1ce17f2641f24ae83637ab66a2cca9c378b9f

## Types of changes

 * New feature (non-breaking change which adds functionality)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Review by moodlezoup — APPROVED** [review]
lgtm, don't forget to update the `@0x/contracts-zero-ex` changelog

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinSushiswap.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "./MixinAdapterAddresses.sol";
+import "./MixinUniswapV2.sol";
+
+contract MixinSushiswap is
+    MixinAdapterAddresses
+{
+    using LibERC20TokenV06 for IERC20TokenV06;
+
+    /// @dev Mainnet address of the `SushiswapV2Router02` contract.
```
Comment (dekz) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinSushiswap.sol:33]:
```suggestion
    /// @dev Mainnet address of the `SushiswapRouter` contract.
```

These never had different versions, so we may as well call them `SUSHISWAP_ROUTER`, incase there ever is a V2 :|

File: contracts/zero-ex/contracts/src/transformers/bridges/BridgeAdapter.sol [hunk 2]
```diff
@@ -117,6 +124,18 @@ contract BridgeAdapter is
                 sellAmount,
                 bridgeData
             );
+        } else if (bridgeAddress == SWERVE_BRIDGE_ADDRESS) {
```
Comment (dekz) [inline contracts/zero-ex/contracts/src/transformers/bridges/BridgeAdapter.sol:127]: I guess we could combine:
```
if (bridgeAddress == CURVE_BRIDGE_ADDRESS || bridgeAddress == SWERVE_BRIDGE_ADDRESS)
```

File: contracts/asset-proxy/contracts/src/bridges/SwerveBridge.sol [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+
+// solhint-disable not-rely-on-time
+// solhint-disable space-after-comma
+contract SwerveBridge is
+    IERC20Bridge,
+    IWallet,
+    DeploymentConstants
+{
+    struct CurveBridgeData {
```
Comment (dorothy-zbornak) [inline contracts/asset-proxy/contracts/src/bridges/SwerveBridge.sol:38]: Worth renaming this?

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinAdapterAddresses.sol [hunk 4]
```diff
@@ -29,11 +29,14 @@ contract MixinAdapterAddresses
...
         address mStableBridge;
         address oasisBridge;
+        address swerveBridge;
+        address sushiswapBridge;
         address uniswapBridge;
         address uniswapV2Bridge;
         // Exchanges
         address kyberNetworkProxy;
         address oasis;
+        address sushiswapV2Router;
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinAdapterAddresses.sol:39]:
```suggestion
        address sushiswapRouter;
```

File: contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinSushiswap.sol [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    IUniswapV2Router02 private immutable SUSHISWAP_V2_ROUTER;
+
+    constructor(AdapterAddresses memory addresses)
+        public
+    {
+        SUSHISWAP_V2_ROUTER = IUniswapV2Router02(addresses.sushiswapV2Router);
+    }
+
+    function _tradeSushiswap(
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinSushiswap.sol:42]: How about we drop this contract, reuse `MixinUniswapV2`, modify it to also store the sushi router, and we can pass in `bool isSushi` to this function? Should make deployments a bit cheaper.

File: contracts/zero-ex/test/transformers/fill_quote_transformer_test.ts [hunk 6]
```diff
@@ -65,10 +65,13 @@ blockchainTests.resets('FillQuoteTransformer', env => {
                 mooniswapBridge: NULL_ADDRESS,
                 mStableBridge: NULL_ADDRESS,
                 oasisBridge: NULL_ADDRESS,
+                sushiswapBridge: NULL_ADDRESS,
+                swerveBridge: NULL_ADDRESS,
                 uniswapBridge: NULL_ADDRESS,
                 uniswapV2Bridge: NULL_ADDRESS,
                 kyberNetworkProxy: NULL_ADDRESS,
                 oasis: NULL_ADDRESS,
+                sushiswapV2Router: NULL_ADDRESS,
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/test/transformers/fill_quote_transformer_test.ts:74]:
```suggestion
                sushiswapRouter: NULL_ADDRESS,
```

File: packages/migrations/src/migration.ts [hunk 7]
```diff
@@ -317,10 +317,13 @@ export async function runMigrationsAsync(
             mooniswapBridge: NULL_ADDRESS,
             mStableBridge: NULL_ADDRESS,
             oasisBridge: NULL_ADDRESS,
+            swerveBridge: NULL_ADDRESS,
+            sushiswapBridge: NULL_ADDRESS,
             uniswapBridge: NULL_ADDRESS,
             uniswapV2Bridge: NULL_ADDRESS,
             kyberNetworkProxy: NULL_ADDRESS,
             oasis: NULL_ADDRESS,
+            sushiswapV2Router: NULL_ADDRESS,
```
Comment (dorothy-zbornak) [inline packages/migrations/src/migration.ts:326]:
```suggestion
            sushiswapRouter: NULL_ADDRESS,
```

**Review by dorothy-zbornak — COMMENTED** [review]
Looking good. What do you think of my suggestion?

**Comment by romainbutteaud** [comment]
Maybe we can add SnowSwap as part of this PR, what do you think @dekz

**Comment by dekz** [comment]
I got you fam

**Pull request closed without merging**